### PR TITLE
Correct `SchemaManagerFunctionalTestCase::testDropsDatabaseWithActiveConnections()` on OracleDB

### LINKED ISSUE #2253: SchemaManagerFunctionalTestCase::testDropsDatabaseWithActiveConnections() fails on Oracle
````
Failed asserting that an array contains 'test_drop_database'.
 /home/deepdiver/Development/ownCloud/doctrine2/dbal/tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php:54

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #2253

### CODE REVIEW

File: tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php [hunk 1]
```diff
@@ -51,10 +52,19 @@ public function testDropsDatabaseWithActiveConnections()
 
         $this->_sm->dropAndCreateDatabase('test_drop_database');
 
-        $this->assertContains('test_drop_database', $this->_sm->listDatabases());
+        $knownDatabases = $this->_sm->listDatabases();
+        if ($this->_conn->getDatabasePlatform() instanceof OraclePlatform) {
```
Comment (Ocramius) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: This indicates a separate issue inside the OraclePlatform IMO. Shouldn't the uppercase conversion happen in there, if upper case is enforced?

Comment (DeepDiver1975) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: well - the table name is uppercased and all tables are returned uppercase - on the api usage side I see no way to know if upper or lowercase is expected.

but maybe I'm not deep enough in this oracle fun :speak_no_evil:

Comment (DeepDiver1975) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: @Ocramius where do we want to go with this? I'm open for any alternative approarch - let me know ... THX

Comment (Ocramius) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: I think this is acceptable, given your feedback. /cc @deeky666

Comment (DeepDiver1975) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: > I think this is acceptable, given your feedback. /cc @deeky666

@deeky666 do you agree? Can we merge this? THX

Comment (DeepDiver1975) [inline tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php]: > @deeky666 do you agree? Can we merge this? THX

ping

**Comment by deeky666** [comment]
The identifier casing issue is not only related to Oracle but I will merge this as is for now. Thanks @DeepDiver1975 !

**Pull request merged**